### PR TITLE
Event Validation

### PR DESCRIPTION
**Pull request by rgoonie**
Created the abstract method `validateEvent` in the `CoffeeShopEvent.java` class and filled in the code for the children classes: `CustomerLeavesEvent` and `NewCustomerWalksInEvent`.

Now why did I have two conditional checks: if null and if it is an instance of. We had a discussion about null pointer exceptions and we always want to make sure that we never get one of those. That's the reason for the null check.

As for the `instanceof` conditional, since the abstracted method uses a type `Object` as the parameter, we want to be 100% sure the data we are checking is the data is the type we want to check. 

Inquires: 
1) Did I do the correct thing making it a `public` method.
2) Where do we want to call the validation: in the constructor of the event or after the class is created?

### CODE REVIEW

File: src/CoffeeShopSimulator/EventBus/Events/CustomerLeavesEvent.java [hunk 1]
```diff
@@ -16,4 +16,26 @@ public CustomerLeavesEvent(Customer customer) {
...
+     * @param data Object to be validated
+     * @return If the data passed is not null and is an instance of Customer
+     */
+    public boolean validateEvent(Object... data){
+        if(data[0] == null){
+            System.out.println("Customer cannot be null!!!");
+            return false;
+        }
+
+        if(! (data[0] instanceof Customer) ){
```
Comment (ynigoreyes) [inline src/CoffeeShopSimulator/EventBus/Events/CustomerLeavesEvent.java]: We should check to see if all data passed to it is the type that we need. We can break this code by `validateEvent(someActualCustomer, null, null)`

Let's take a look at [this](https://stackoverflow.com/questions/5307172/check-if-all-items-are-the-same-in-a-list)

File: src/CoffeeShopSimulator/EventBus/Events/NewCustomerWalksInEvent.java [hunk 2]
```diff
@@ -16,4 +16,26 @@ public NewCustomerWalksInEvent(Customer customer) {
...
+     * @param data Object to be validated
+     * @return If the data passed is not null and is an instance of Customer
+     */
+    public boolean validateEvent(Object... data){
+        if(data[0] == null){
+            System.out.println("Customer cannot be null!!!");
+            return false;
+        }
+
+        if(! (data[0] instanceof Customer) ){
```
Comment (ynigoreyes) [inline src/CoffeeShopSimulator/EventBus/Events/NewCustomerWalksInEvent.java]: Same thing as the first class

**Review by ynigoreyes — CHANGES_REQUESTED** [review]

File: src/CoffeeShopSimulator/EventBus/Events/CustomerLeavesEvent.java (hunk 1 above)
Comment (rgoonie) [reply src/CoffeeShopSimulator/EventBus/Events/CustomerLeavesEvent.java]: Just for confirmation and to make sure I put this right. For this validation and the `NewCustomerWalksInEvent()` we want to make sure all the object we pass are customers correct? Or do we want to make sure that all the objects are the same customer?

Comment (ynigoreyes) [reply src/CoffeeShopSimulator/EventBus/Events/CustomerLeavesEvent.java]: all data are customers, not necessarily the same one, but they are all of type Customer

**Review by ynigoreyes — APPROVED** [review]

**Pull request merged**